worker/pkg/workflows/datasync/activities/activities_test.go

```diff
@@ -310,6 +310,154 @@ func Test_buildProcessorMutation(t *testing.T) {
 	})
 	assert.Error(t, err)
 	assert.Empty(t, output)
+
+}
+
+func Test_buildProcessorConfigJavascript(t *testing.T) {
+	mockTransformerClient := mgmtv1alpha1connect.NewMockTransformersServiceClient(t)
+	mockJobClient := mgmtv1alpha1connect.NewMockJobServiceClient(t)
+	mockConnectionClient := mgmtv1alpha1connect.NewMockConnectionServiceClient(t)
+
+	pgcache := map[string]pg_queries.DBTX{
+		"fake-prod-url":  pg_queries.NewMockDBTX(t),
+		"fake-stage-url": pg_queries.NewMockDBTX(t),
+	}
+	pgquerier := pg_queries.NewMockQuerier(t)
+	mysqlcache := map[string]mysql_queries.DBTX{}
+	mysqlquerier := mysql_queries.NewMockQuerier(t)
+
+	bbuilder := newBenthosBuilder(pgcache, pgquerier, mysqlcache, mysqlquerier, mockJobClient, mockConnectionClient, mockTransformerClient)
+	ctx := context.Background()
+
+	mockTransformerClient.On(
+		"GetUserDefinedTransformerById",
+		mock.Anything,
+		connect.NewRequest(&mgmtv1alpha1.GetUserDefinedTransformerByIdRequest{
+			TransformerId: "456",
+		}),
+	).Return(connect.NewResponse(&mgmtv1alpha1.GetUserDefinedTransformerByIdResponse{
+		Transformer: &mgmtv1alpha1.UserDefinedTransformer{
+			Id:          "456",
+			Name:        "stage",
+			Description: "description",
+			DataType:    "string",
+			Source:      "javascript",
+			Config: &mgmtv1alpha1.TransformerConfig{
+				Config: &mgmtv1alpha1.TransformerConfig_JavascriptConfig{
+					JavascriptConfig: &mgmtv1alpha1.Javascript{
+						Code: `(()=>{var payload = benthos.v0_msg_as_structured();payload.value+=" helloee";})();`,
+					},
+				},
+			},
+		},
+	}), nil)
+
+	res, err := bbuilder.buildProcessorConfig(ctx, []*mgmtv1alpha1.JobMapping{
+		{Schema: "public", Table: "users", Column: "id", Transformer: &mgmtv1alpha1.JobMappingTransformer{Source: "javascript", Config: &mgmtv1alpha1.TransformerConfig{
+			Config: &mgmtv1alpha1.TransformerConfig_UserDefinedTransformerConfig{
+				UserDefinedTransformerConfig: &mgmtv1alpha1.UserDefinedTransformerConfig{
+					Id: "456",
+				},
+			},
+		}}},
+	})
+
+	assert.NoError(t, err)
+	assert.Equal(t, res.Javascript.Code, `(()=>{var payload = benthos.v0_msg_as_structured();payload.id+=" helloee";})();`)
```

Comment:
How does this work if there are multiple job mappings that define a javascript processor? Would be great to see a test for the expected output here.